src/lib/sqle/dictionary/dictionary.service.ts

```diff
@@ -0,0 +1,829 @@
+
+import { VantageQueryService, IQueryResultSet, IQueryPayload, ISQLEConnection } from '../query/query.service';
+
+export const sysDatabases: string[] = [
+  'DBC',
+  'dbcmngr',
+  'SQLJ',
+  'SystemFe',
+  'SysAdmin',
+  'SYSBAR',
+  'SYSJDBC',
+  'SYSLIB',
+  'SYSUDTLIB',
+  'TDMaps',
+  'TD_SERVER_DB',
+  'TD_SYSFNLIB',
+  'TD_SYSXML',
+  'Sys_Calendar',
+];
+
+export interface IDictionarySystem {
+  hostname?: string;
+  password?: string;
+  port?: number;
+  system_id?: string;
+  last_run?: string;
+  interval_minutes?: number;
+  enabled?: boolean;
+  last_attempt?: string;
+  in_collection?: boolean;
+  username?: string;
+}
+
+export interface IDictionaryDatabase {
+  parent_name?: string;
+  size_bytes?: number;
+  system_id?: string;
+  name?: string;
+  id?: string;
+  type?: string;
+}
+
+export interface IDictionaryDatabaseObject {
+  system_id?: string;
+  database_id?: string;
+  id?: string;
+  name?: string;
+  size_bytes?: number;
+  type?: string;
+  create_text?: string;
+}
+
+export interface IDictionaryTableColumn {
+  system_id?: string;
+  database_id?: string;
+  table_id?: string;
+  id?: string;
+  name?: string;
+}
+
+/**
+ * FunctionType
+ * A - Aggregate
+ * B - Aggregate and statistical
+ * C - Contract function
+ * D - External stored procedure
+ * F - Scalar
+ * H - User defined method
+ * I - Internal type method
+ * L - Table operator
+ * R - Table Function
+ * S - Statistical
+ */
+
+/**
+ * TableKind for dbc.tablesvx;
+ * A - Aggregate function
+ * B - Combined aggregate and ordered analytical function
+ * C - Table operator parser contract function
+ * D - JAR
+ * E - External Stored Procedure
+ * F - Standard Function
+ * G - Trigger
+ * H - Instance or contructor method
+ * I - Join Index
+ * J - Journal
+ * K - Foreign Server Object
+ * L - User defined table operator
+ * M - Macro
+ * N - Hash index
+ * O - Table with no primary index and no partitioning
+ * P - Stored Procedure
+ * Q - Queue table
+ * R - Table function
+ * S - Ordered Analytical function
+ * T - table with a primary index or primary AMP index, partitioning or both. Or a partioned table with NoPI;
+ * U - User defined type
+ * V - View
+ * X - Authorization
+ * Y - GLOP set
+ * Z - UIF - User Installed File
+ * 1 - A DATASET schema object created by CREATE SCHEMA
+ * 2 - Functon alias object
+ */
+
+/**
+ * dbc.externalSPs
+ * dbc.sessionInfovx
+ */
+
+@Injectable()
+export class VantageDictionaryService {
+  constructor(private _queryService: VantageQueryService) {}
+
+  async getFrequencyAnalysis(
+    connection: ISQLEConnection,
+    database: string,
+    table: string,
+    column: string,
+  ): Promise<any> {
+    const requests: any[] = [];
+    const session: any = await this._queryService.createSession(connection).toPromise();
+    const payload: IQueryPayload = {
+      query: `
+        call demouser.td_analyze('frequency', 'database=${database};tablename=${table};columns=${column};');
```

Comment:
I think demouser should not be reference here? Not sure what to substitute.

Reply:
We can probably remove these methods, they were supposed to be used on the profiling functions, but they are still not part of vantage.. so they are leftovers of the PoC.